backend/battles/forms.py

```diff
@@ -37,7 +39,30 @@ def save(self, commit=True):
         pokemon_3 = self.cleaned_data["pokemon_3"]
 
         team = (pokemon_1, pokemon_2, pokemon_3)
-
         instance = Team.objects.create(trainer=trainer, battle=self.initial["battle"])
         instance.team.set(team)
         return instance
+
+    def clean(self):
+
+        cleaned_data = super().clean()
+
+        pokemon_1 = cleaned_data.get("pokemon_1")
+        pokemon_2 = cleaned_data.get("pokemon_2")
+        pokemon_3 = cleaned_data.get("pokemon_3")
+
+        team = (pokemon_1, pokemon_2, pokemon_3)
+
+        for pokemon in team:
+            if pokemon < 0 or pokemon > POKE_API_LIMIT:
+                raise forms.ValidationError("Choose a valid pokemon")
+
+        if duplicate_pokemon(team):
+            raise forms.ValidationError("Your team has duplicates, please use unique ids")
+
+        if not pokemon_attr_exceeds_limit(team):
```

Comment:
In the name `pokemon_attr_exceeds_limit`, for what `attr` stands for? Shouldn't this be `pokemon_team_exceeds_limit`?

Reply:
Makes sense. I also thought of `team_stats_exceeds_limits`, but i think in that logic the word 'stats' seems vague.